File: Data Structures & Algorithms/linked-list-cycle-detection/submission-7.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        f=head.next 
        s=head

        while f and f.next:
            if f==s:
                return True
            f=f.next.next
            s=s.next

        return False

File: Data Structures & Algorithms/reverse-a-linked-list/submission-7.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode left = null;
        ListNode right = head;
        while(right != null) {
            ListNode temp = right.next;
            //reverse pointer
            right.next = left;
            //geser left pointer
            left = right;
            //geser right pointer
            right = temp;
        }
        return left;
    }
}
